bootloader/start64.c

```diff
@@ -67,16 +88,53 @@ uint64_t relocate_kernel (uint32_t mboot_struct_addr) {
   uint32_t bss_end_addr = kernel[i+6];
   uint32_t entry_addr = kernel[i+7];
 
+  /* Find the last used memory address */
+  uint64_t last_used = bss_end_addr;
+  if (last_used < (uint64_t)&mboot_struct[7]) {
+    last_used = (uint64_t)&mboot_struct[7];
+  }
+  if (last_used < (uint64_t)&mod_list[mboot_struct[5]]) {
+    last_used = (uint64_t)&mod_list[mboot_struct[5]];
+  }
+  for (uint16_t i = 0; i < mboot_struct[5]; i++) {
+    if (last_used < (uint64_t)mod_list[i].end) {
+      last_used = (uint64_t)mod_list[i].end;
+    }
+    // TODO: check string locations as well
```

Comment:
IMO this shouldn't be accepted if this isn't fixed first (and the string copying)